contracts/src/core/interfaces/IEigenDADirectory.sol

```diff
@@ -48,72 +50,39 @@ interface IEigenDAConfigRegistry {
-    /// @dev Reverts if the entry already exists.
-    function addConfigBytes(string memory name, bytes memory value, string memory extraInfo) external;
-
-    /// @notice Replaces an existing bytes32 configuration entry in the registry by name, along with optional extra info.
-    /// @dev Reverts if the entry does not exist.
-    function replaceConfigBytes32(string memory name, bytes32 value, string memory extraInfo) external;
-
-    /// @notice Replaces an existing bytes configuration entry in the registry by name, along with optional extra info.
-    /// @dev Reverts if the entry does not exist.
-    function replaceConfigBytes(string memory name, bytes memory value, string memory extraInfo) external;
-
-    /// @notice Removes an existing bytes32 configuration entry from the registry by name.
-    /// @dev Reverts if the entry does not exist.
-    function removeConfigBytes32(string memory name) external;
+    function addConfigBytes32(string memory name, uint256 activationKey, bytes32 value) external;
 
-    /// @notice Removes an existing bytes configuration entry from the registry by name.
-    /// @dev Reverts if the entry does not exist.
-    function removeConfigBytes(string memory name) external;
+    function addConfigBytes(string memory name, uint256 activationKey, bytes memory value) external;
 
-    /// @notice Gets the bytes32 configuration entry by name.
-    function getConfigBytes32(string memory name) external view returns (bytes32);
+    function getNumCheckpointsBytes32(bytes32 key) external view returns (uint256);
 
-    /// @notice Gets the bytes configuration entry by name.
-    function getConfigBytes(string memory name) external view returns (bytes memory);
+    function getNumCheckpointsBytes(bytes32 key) external view returns (uint256);
 
-    /// @notice Gets the extra info associated with a bytes32 configuration entry by name.
-    function getConfigBytes32ExtraInfo(string memory name) external view returns (string memory);
+    function getConfigBytes32(bytes32 key, uint256 index) external view returns (bytes32);
 
-    /// @notice Gets the extra info associated with a bytes configuration entry by name.
-    function getConfigBytesExtraInfo(string memory name) external view returns (string memory);
+    function getConfigBytes(bytes32 key, uint256 index) external view returns (bytes memory);
 
-    /// @notice Gets the bytes32 configuration entry by keccak256 hash of the name.
-    function getConfigBytes32(bytes32 key) external view returns (bytes32);
+    function getActivationKeyBytes32(bytes32 key, uint256 index) external view returns (uint256);
 
-    /// @notice Gets the bytes configuration entry by keccak256 hash of the name.
-    function getConfigBytes(bytes32 key) external view returns (bytes memory);
+    function getActivationKeyBytes(bytes32 key, uint256 index) external view returns (uint256);
```

Comment:
What's the plan for how to use these offchain?
Afaiu the getter function we would want to use is something like the router, where instead of passing an index, we pass an RBN (or timestamp), and the getter function walks the list and find the active value at that "referenceKey".

Reply:
This also randomly made me wonder.. do we really need to checkpoint by block numbers at all? Or could we only use timestamps like cafe does and then any offchain user just needs to convert the RBN to a timestamp before calling the configRegistry? This would standardize the config and then we could use "activation/reference time" instead of "activationKey". Thoughts? cc @litt3